File: java/adaptador/midia-player/src/main/java/edu/uea/adapter/MP4Adapter.java
package edu.uea.adapter;

public class MP4Adapter implements MediaPlayer {
    private MP4Player mp4Player;

    public MP4Adapter() {
        this.mp4Player = new MP4Player();
    }

    @Override
    public void play(String audioType, String fileName) {
        if (audioType.equalsIgnoreCase("mp4")) {
            mp4Player.show(fileName);
        }
    }

}

File: python/adaptador/adapter/MP4Adapter.py
from .MediaPlayer import MediaPlayer
from .MP4Player import MP4Player 

# Adapter Class
class MP4Adapter(MediaPlayer):
    def __init__(self):
        self.mp4Player = MP4Player()

    def play(self, audio_type, file_name):
        if audio_type.lower() == "mp4":
            self.mp4Player.show(file_name)